testing/unittests/jlibtests.cpp

```diff
@@ -3489,211 +3689,116 @@ class PTreeSerializationTest : public CppUnit::TestFixture
+        CPPUNIT_ASSERT(memcmp(serializedData.toByteArray(), streamSerializedData.toByteArray(), serializedData.length()) == 0);
 
-        // Large binary data (over PTREE_COMPRESS_THRESHOLD to trigger compression)
-        // PTREE_COMPRESS_THRESHOLD is typically 4KB based on the comment in the existing test
-        const size_t largeSize = 8 * 1024; // 8KB to ensure compression
-        largeBinary.ensureCapacity(largeSize);
-        for (size_t i = 0; i < largeSize; i++)
-            largeBinary.append((byte)(i % 256));
-        tree->setPropBin("largeBinary", largeBinary.length(), largeBinary.toByteArray());
+        // Test deserialize() method
+        MemoryBuffer deserializeBuffer;
+        deserializeBuffer.append(serializedData.length(), serializedData.toByteArray());
+        Owned<IPropertyTree> deserializedTree = createPTree();
+        CCycleTimer timer;
+        deserializedTree->deserialize(deserializeBuffer);
+        __uint64 deserializeElapsedNs = timer.elapsedNs();
+
+        // Test deserializeFromStream() method
+        MemoryBuffer streamBuffer;
+        streamBuffer.append(streamSerializedData.length(), streamSerializedData.toByteArray());
+        Owned<IBufferedSerialInputStream> in = createBufferedSerialInputStream(streamBuffer);
+        Owned<IPropertyTree> deserializedFromStreamTree = createPTree();
+        timer.reset();
+        deserializedFromStreamTree->deserializeFromStream(*in);
+        __uint64 deserializeFromStreamElapsedNs = timer.elapsedNs();
 
-        // Add nested elements with binary data
-        IPropertyTree *subTree = tree->addPropTree("subElement");
-        subTree->setProp("subProp", "subValue");
+        // Verify both deserialized trees are equivalent to the original
+        CPPUNIT_ASSERT(areMatchingPTrees(originalTree, deserializedTree));
+        CPPUNIT_ASSERT(areMatchingPTrees(originalTree, deserializedFromStreamTree));
 
-        MemoryBuffer nestedBinary;
-        const char *nestedData = "Nested binary content with some repetitive data for compression testing";
-        // Repeat the data to make it larger and more compressible
-        for (int i = 0; i < 100; i++)
-            nestedBinary.append(strlen(nestedData), nestedData);
-        subTree->setPropBin("nestedBinary", nestedBinary.length(), nestedBinary.toByteArray());
+        // Convert to milliseconds for display
+        double deserializeTimeMs = deserializeElapsedNs / 1e6;
+        double deserializeFromStreamTimeMs = deserializeFromStreamElapsedNs / 1e6;
 
-        return tree.getClear();
+        DBGLOG("Deserialization timing results for %s:", testName);
+        DBGLOG("  deserialize() time: %.6f ms", deserializeTimeMs);
+        DBGLOG("  deserializeFromStream() time: %.6f ms", deserializeFromStreamTimeMs);
+        DBGLOG("  Performance ratio (deserializeFromStream/deserialize): %.6f", deserializeFromStreamTimeMs / deserializeTimeMs);
```

Comment:
Not strictly related to this PR, but it would also be very interesting to include the time to serialize/deserialize to text, and the size comparison.